src/Swift.Bindings/tests/CryptoKit/Foundation.Source.cs

```diff
@@ -0,0 +1,274 @@
+
+        [DllImport(Path, EntryPoint = "swift_getWitnessTable")]
+        [UnmanagedCallConv(CallConvs = [ typeof(CallConvSwift) ])]
+        public static unsafe extern void* PInvoke_Swift_GetWitnessTable(void* conformanceDescriptor, void* typeMetadata, void* instantiationArgs);
+
+        [DllImport(Path, EntryPoint = "$s10Foundation4DataVMa")]
+        [UnmanagedCallConv(CallConvs = [ typeof(CallConvSwift) ])]
+        public static unsafe extern void* PInvoke_Data_GetMetadata();
+    }
+
+    /// <summary>
+    /// Swift runtime helper methods in C#.
+    /// </summary>
+    public static class Runtime
+    {
+        /// <summary>
+        /// https://github.com/apple/swift/blob/main/include/swift/ABI/MetadataValues.h#L117
+        /// </summary>
+        [Flags]
+        public enum ValueWitnessFlags
+        {
+            AlignmentMask = 0x0000FFFF,
+            IsNonPOD = 0x00010000,
+            IsNonInline = 0x00020000,
+            HasSpareBits = 0x00080000,
+            IsNonBitwiseTakable = 0x00100000,
+            HasEnumWitnesses = 0x00200000,
+            Incomplete = 0x00400000,
+        }
+
+        /// <summary>
+        /// See https://github.com/apple/swift/blob/main/include/swift/ABI/ValueWitness.def
+        /// </summary>
+        [StructLayout (LayoutKind.Sequential)]
+        public ref struct ValueWitnessTable
+        {
+            public IntPtr InitializeBufferWithCopyOfBuffer;
+            public IntPtr Destroy;
+            public IntPtr InitWithCopy;
+            public IntPtr AssignWithCopy;
+            public IntPtr InitWithTake;
+            public IntPtr AssignWithTake;
+            public IntPtr GetEnumTagSinglePayload;
+            public IntPtr StoreEnumTagSinglePayload;
+            private IntPtr _Size;
+            private IntPtr _Stride;
+            public ValueWitnessFlags Flags;
+            public uint ExtraInhabitantCount;
+            public int Size => _Size.ToInt32();
+            public int Stride => _Stride.ToInt32();
+            public int Alignment => (int)((Flags & ValueWitnessFlags.AlignmentMask) + 1);
+            public bool IsNonPOD => Flags.HasFlag (ValueWitnessFlags.IsNonPOD);
+            public bool IsNonBitwiseTakable => Flags.HasFlag (ValueWitnessFlags.IsNonBitwiseTakable);
+            public bool HasExtraInhabitants => ExtraInhabitantCount != 0;
+        }
+
+        public static unsafe void* GetMetadata<T>(T type) where T: ISwiftObject
+        {
+            return T.Metadata;
+        }
+
+        public static unsafe void* GetValueWitnessTable(void* metadata)
+        {
+            void* valueWitnessTable = (void*)Marshal.ReadIntPtr((IntPtr)metadata, -IntPtr.Size);
+            return valueWitnessTable;
+        }
+
+        public static unsafe void* GetConformanceDescriptor(string symbol)
+        {
+            IntPtr handle = IntPtr.Zero;
+            try
+            {
+                handle = NativeLibrary.Load(Foundation.Path);
```

Comment:
This will only work for conformance descriptors that are in Foundation - there will be other modules that have conformance descriptors. We should open an issue on this so it doesn't get lost.

Reply:
Created a tracking issue: https://github.com/dotnet/runtimelab/issues/2705